### DISCUSSION TITLE
SudoLang = Potential LLM to LLM Communication Technique -- A Programming Language  for LLMs

### DISCUSSION THREAD

**Discussion by ScottzCodez** [daveshap/ACE_Framework#34] [General]
This could reduce the Token counts of all the LLM communications, and provide a powerful tool set, and sort of structure to the Dialog.

https://medium.com/javascript-scene/sudolang-a-powerful-pseudocode-programming-language-for-llms-d64d42aa719b
https://github.com/paralleldrive/sudolang-llm-support

https://www.google.com/search?q=SudoLang

**Comment by DataBassGit** [comment]
You can absolutely use SudoLang in an implementation, however, you have to send the sudolang priming prompt every time you want to use it. Unlike ChatGPT, calling GPT-4 from the API does not include all chat history automatically. I see powerful potential for SudoLang in cognitive architecture, but with the caveat that it is not as consistent as hard coding the function.

That said, the ACE Framework is not specific to any tech. Language Model, Database, Communication Protocols, Structured Data, those are all up to the developer.

**Comment by CognitiveCodes** [comment]
Maybe you'll be interested in my project?
NeuralGPT - a functional LLM<->LLM communiction system
https://github.com/daveshap/ACE_Framework/discussions/11
